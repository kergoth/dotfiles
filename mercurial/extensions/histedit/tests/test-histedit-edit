#!/bin/sh

. "$TESTDIR/histedit-helpers.sh"

cat >> $HGRCPATH <<EOF
[extensions]
hgext.graphlog=
EOF

echo "histedit=$(echo $(dirname $(dirname $0)))/hg_histedit.py" >> $HGRCPATH

EDITED=`pwd`/editedhistory
cat > $EDITED <<EOF
pick 177f92b77385 c
pick 055a42cdd887 d
edit e860deea161a e
pick 652413bf663e f
EOF
initrepo ()
{
    hg init r
    cd r
    for x in a b c d e f ; do
        echo $x > $x
        hg add $x
        hg ci -m $x
    done
}

initrepo

echo % log before edit
hg log --graph

echo % edit the history
HGEDITOR="cat $EDITED > " hg histedit 177f92b77385 2>&1 | fixbundle

echo % edit the revision
echo a > e
HGEDITOR='echo "foobaz" > ' hg histedit --continue 2>&1 | fixbundle

hg log --graph

echo '% contents of e:'
hg cat e

cat > $EDITED <<EOF
edit c38516e9ed62 f
EOF
HGEDITOR="cat $EDITED > " hg histedit tip 2>&1 | fixbundle
hg status
HGEDITOR='true' hg histedit --continue
hg status

echo % log after edit
hg log --limit 1

echo "% say we'll change the message, but don't."
cat > ../edit.sh <<EOF
#!/bin/sh
cat \$1 | sed s/pick/mess/ > tmp
mv tmp \$1
EOF
chmod +x ../edit.sh
HGEDITOR="../edit.sh" hg histedit tip 2>&1 | fixbundle
hg status
hg log --limit 1

echo % modify the message
cat > $EDITED <<EOF
mess c38516e9ed62 f
EOF
HGEDITOR="cat $EDITED > " hg histedit tip 2>&1 | fixbundle
hg status
hg log --limit 1

echo % rollback should not work after a histedit
hg rollback

echo % EOF
