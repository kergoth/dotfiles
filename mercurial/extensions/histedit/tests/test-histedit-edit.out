% log before edit
@  changeset:   5:652413bf663e
|  tag:         tip
|  user:        test
|  date:        Thu Jan 01 00:00:00 1970 +0000
|  summary:     f
|
o  changeset:   4:e860deea161a
|  user:        test
|  date:        Thu Jan 01 00:00:00 1970 +0000
|  summary:     e
|
o  changeset:   3:055a42cdd887
|  user:        test
|  date:        Thu Jan 01 00:00:00 1970 +0000
|  summary:     d
|
o  changeset:   2:177f92b77385
|  user:        test
|  date:        Thu Jan 01 00:00:00 1970 +0000
|  summary:     c
|
o  changeset:   1:d2ae7f538514
|  user:        test
|  date:        Thu Jan 01 00:00:00 1970 +0000
|  summary:     b
|
o  changeset:   0:cb9a9f314b8b
   user:        test
   date:        Thu Jan 01 00:00:00 1970 +0000
   summary:     a

% edit the history
0 files updated, 0 files merged, 2 files removed, 0 files unresolved
abort: Make changes as needed, you may commit or record as needed now.
When you are finished, run hg histedit --continue to resume.
% edit the revision
0 files updated, 0 files merged, 0 files removed, 0 files unresolved
0 files updated, 0 files merged, 0 files removed, 0 files unresolved
@  changeset:   5:c38516e9ed62
|  tag:         tip
|  user:        test
|  date:        Thu Jan 01 00:00:00 1970 +0000
|  summary:     f
|
o  changeset:   4:1da62d13177d
|  user:        test
|  date:        Thu Jan 01 00:00:00 1970 +0000
|  summary:     foobaz
|
o  changeset:   3:055a42cdd887
|  user:        test
|  date:        Thu Jan 01 00:00:00 1970 +0000
|  summary:     d
|
o  changeset:   2:177f92b77385
|  user:        test
|  date:        Thu Jan 01 00:00:00 1970 +0000
|  summary:     c
|
o  changeset:   1:d2ae7f538514
|  user:        test
|  date:        Thu Jan 01 00:00:00 1970 +0000
|  summary:     b
|
o  changeset:   0:cb9a9f314b8b
   user:        test
   date:        Thu Jan 01 00:00:00 1970 +0000
   summary:     a

% contents of e:
a
0 files updated, 0 files merged, 1 files removed, 0 files unresolved
abort: Make changes as needed, you may commit or record as needed now.
When you are finished, run hg histedit --continue to resume.
A f
0 files updated, 0 files merged, 0 files removed, 0 files unresolved
% log after edit
changeset:   5:c38516e9ed62
tag:         tip
user:        test
date:        Thu Jan 01 00:00:00 1970 +0000
summary:     f

% say we'll change the message, but don't.
0 files updated, 0 files merged, 1 files removed, 0 files unresolved
0 files updated, 0 files merged, 0 files removed, 0 files unresolved
changeset:   5:c38516e9ed62
tag:         tip
user:        test
date:        Thu Jan 01 00:00:00 1970 +0000
summary:     f

% modify the message
0 files updated, 0 files merged, 1 files removed, 0 files unresolved
0 files updated, 0 files merged, 0 files removed, 0 files unresolved
changeset:   5:4d6a10bcf3e3
tag:         tip
user:        test
date:        Thu Jan 01 00:00:00 1970 +0000
summary:     mess c38516e9ed62 f

% rollback should not work after a histedit
no rollback information available
% EOF
